pkg/controllers/vdb/httpserverctrl_reconciler.go

```diff
@@ -0,0 +1,106 @@
+/*
+ (c) Copyright [2021-2023] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vdb
+
+import (
+	"context"
+	"fmt"
+
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/cmds"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	"github.com/vertica/vertica-kubernetes/pkg/names"
+	"github.com/vertica/vertica-kubernetes/pkg/paths"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// HTTPServerCtrlReconciler will start the http server if needed.
+type HTTPServerCtrlReconciler struct {
+	VRec    *VerticaDBReconciler
+	Vdb     *vapi.VerticaDB // Vdb is the CRD we are acting on.
+	PRunner cmds.PodRunner
+	PFacts  *PodFacts
+}
+
+func MakeHTTPServerCtrlReconciler(vdbrecon *VerticaDBReconciler,
+	vdb *vapi.VerticaDB, prunner cmds.PodRunner, pfacts *PodFacts) controllers.ReconcileActor {
+	return &HTTPServerCtrlReconciler{
+		VRec:    vdbrecon,
+		Vdb:     vdb,
+		PRunner: prunner,
+		PFacts:  pfacts,
+	}
+}
+
+func (h *HTTPServerCtrlReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctrl.Result, error) {
+	// Early out if the http service isn't enabled.
+	if !h.doHTTPStart(true) {
+		return ctrl.Result{}, nil
+	}
+
+	if err := h.PFacts.Collect(ctx, h.Vdb); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	for _, pod := range h.PFacts.Detail {
+		if pod.isHTTPServerRunning {
+			continue
+		}
+		// Only start the http server for pods that have been added to a database
+		if !pod.upNode {
+			h.VRec.Log.Info("Not all pods are up.  Need to requeue httpserverctrl reconciler.", "pod", pod.name)
```

Comment:
We aren't doing a requeue. Should we just log the pod?
```suggestion
			h.VRec.Log.Info("Skipping http server start on pod because vertica is not running", "pod", pod.name)
```